src/dbnode/storage/stats/query_stats.go

```diff
@@ -49,6 +49,15 @@ type QueryStats interface {
 	Stop()
 }
 
+// QueryStatsConfig holds configuration for how a tracker should handle query stats.
+type QueryStatsConfig struct {
+	// MaxDocs limits how many recently queried max
+	// documents are allowed before queries are abandoned.
+	MaxDocs *int64
```

Comment:
I think this would be better as an `int64`, with 0 being the no limit case rather than the pointer?

Reply:
Agreed, probably better to just have them absolute value types